src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -345,6 +346,59 @@ SIMPLE_MATCHER_PASS_DEFINITION(EliminateGather,
                                ov::op::v7::Gather,
                                ov::op::v8::Gather);
 
+pass::EliminateReduceReshape::EliminateReduceReshape() {
+    MATCHER_SCOPE(EliminateReduceReshape);
+    using namespace pass::pattern;
+    auto axes = wrap_type<ov::op::v0::Constant>();
+    auto reduce_pattern = wrap_type<ov::op::util::ReductionBase>({any_input(), axes});
+    auto requested_shape_pattern = wrap_type<ov::op::v0::Constant>();
+    auto reshape_pattern =
+        wrap_type<ov::op::v1::Reshape>({reduce_pattern, requested_shape_pattern}, consumers_count(1));
+
+    matcher_pass_callback callback = [=](Matcher& m) {
+        auto pattern_map = m.get_pattern_map();
+        auto reshape_node = m.get_match_root();
+        auto reduce_node = pattern_map.at(reduce_pattern);
+
+        auto reshape = ov::as_type_ptr<ov::op::v1::Reshape>(reshape_node);
+        auto reduce = ov::as_type_ptr<ov::op::util::ReductionBase>(reduce_node);
+        if (!reshape || !reduce) {
+            return false;
+        }
+
+        auto in_rank = reshape->get_input_partial_shape(0).rank();
+        auto out_rank = reshape->get_output_partial_shape(0).rank();
+        if (in_rank.is_dynamic() || out_rank.is_dynamic() || in_rank.get_length() != out_rank.get_length()) {
+            return false;
+        }
+
+        auto requested_shape = ov::as_type_ptr<ov::op::v0::Constant>(pattern_map.at(requested_shape_pattern));
+        if (!requested_shape) {
```

Comment:
Is it possible to exit here as `requested_shape_pattern` check that this input is Constant?

Reply:
I would prefer to have nullptr check after dynamic cast